daml-foundations/daml-ghc/src/DA/Daml/GHC/Compiler/Convert.hs

```diff
@@ -240,6 +240,27 @@ convertInt64 x
     | otherwise =
         unsupported "Int literal out of bounds" (negate x)
 
+convertRational :: Integer -> Integer -> ConvertM LF.Expr
+convertRational num denom
+ =
+    -- the denominator needs to be a divisor of 10^10.
+    -- num % denom * 10^10 needs to fit within a 128bit signed number.
+    if | 10 ^ maxPrecision `mod` denom == 0 &&
+             lowerBound128Bit < (r * 10 ^ (10 :: Integer)) &&
+             r * 10 ^ maxPrecision <= upperBound128Bit ->
+           pure $ EBuiltin $ BEDecimal $ fromRational r
```

Comment:
I think we should use `MkFixed` instead of `fromRational`. I'd like to really make sure there's no implicit rounding.